clients/drcachesim/tools/schedule_stats.cpp

```diff
@@ -359,6 +384,27 @@ schedule_stats_t::parallel_shard_memref(void *shard_data, const memref_t &memref
             shard->cur_segment_instrs = 0;
         }
         shard->direct_switch_target = INVALID_THREAD_ID;
+        if (shard->last_syscall_number >= 0 && !shard->stream->is_record_kernel()) {
+            if (shard->stream->get_version() < TRACE_ENTRY_VERSION_FREQUENT_TIMESTAMPS) {
+                // Legacy versions do not have the timestamps to compute latencies.
+            } else {
+                assert(shard->pre_syscall_timestamp > 0);
+                assert(shard->post_syscall_timestamp > 0);
+                histogram_interface_t *hist_ptr = find_or_add_histogram(
+                    shard->counters.sysnum_noswitch_latency, shard->last_syscall_number,
```

Comment:
The passing unit test probably means it's WAI, but it isn't clear to me that control flows here only when there was no switch.

L373 is for the case where a switch happened, but this part isn't in the else-block.

Reply:
The else of L373 is for every single record not after a switch, where we want only to be after a syscall.  I added this comment at line 388:

            // Since we clear last_syscall_number on detecting a context switch prior to
            // here, if it's still set and we've found a regular instruction, we just
            // completed a system call but did not switch.